boa_parser/src/parser/statement/mod.rs

```diff
@@ -283,6 +294,25 @@ where
             let item =
                 StatementListItem::new(self.allow_yield, self.allow_await, self.allow_return)
                     .parse(cursor, interner)?;
+
+            if directive_prologues {
+                if let ast::StatementListItem::Statement(ast::Statement::Expression(
+                    ast::Expression::Literal(ast::expression::literal::Literal::String(string)),
+                )) = &item
+                {
+                    if interner.resolve_expect(*string).join(
+                        |s| s == "use strict",
+                        |g| g == utf16!("use strict"),
+                        true,
+                    ) {
+                        cursor.set_strict_mode(true);
+                        strict = true;
+                    }
+                } else {
+                    directive_prologues = false;
+                }
+            }
```

Comment:
I think I found a small bug. On this PR, this will throw:

```javascript
function f(){
    "a";
    ;
    "use strict";
    let public = 5;
}
```

However, I ran that on Spidermonkey and it successfully runs:

![image](https://user-images.githubusercontent.com/38230983/211948871-62f6479d-11d1-4ece-9815-cc4beed56f8c.png)

This means the parser should stop considering string expressions as directives if there's an empty statement between them.

Reply:
I don't know why we are skipping consecutive semicolons though, the parser should parse that as a `Statement::Empty` statement...

Reply:
Huh, test262 doesn't have a test to assert this. Can you add my snippet to our suite?

Reply:
Yeah. I'll have to fix some other places where we do not parse semicolon correctly. That's why some of our tests are now breaking...

Reply:
Hm yeah, the fix seems less trivial than I thought...
If you prefer, let's split this PR into two parts to avoid polluting this one with unrelated fixes.

Reply:
Yeah I'm working on it now and there are some things in the for loop parsing that need to be fixed. I will split it into a new PR and push the initial changes to this one.